src/clr-angular/forms/select/select.spec.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Component} from "@angular/core";
+
+import {TestContext} from "../../data/datagrid/helpers.spec";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+
+import {ClrSelect} from "./select";
+
+@Component({
+    template: `
+        <clr-select>
+            Test
+        </clr-select>
+    `
+})
+class TestComponent {}
+
+export default function(): void {
+    describe("Select Component", function() {
+        let context: TestContext<ClrSelect, TestComponent>;
+        let ifOpenService: IfOpenService;
+
+        beforeEach(function() {
+            context = this.create(ClrSelect, TestComponent, [IfOpenService]);
+            ifOpenService = context.getClarityProvider(IfOpenService);
+        });
+
+        describe("Typescript API", function() {
+            it("provides a method to toggle the popover on click", () => {
+                expect(ifOpenService.open).toBeUndefined();
+
+                context.clarityDirective.onClick(new MouseEvent("click"));
+
+                expect(open).not.toBe(true);
+
+                context.clarityDirective.onClick(new MouseEvent("click"));
+
+                expect(open).not.toBe(false);
+            });
+
+            it("provides a method to close the popover on focusout", () => {
+                expect(ifOpenService.open).toBeUndefined();
+
+                context.clarityDirective.onFocusOut();
+                expect(ifOpenService.open).toBe(false);
```

Comment:
That's a very strange test, going from `undefined` to `false`.  But ok.

... I guess it'd be a bitt better if it _started_ with `ifOpenService.open = true` instead of the first `expect`.

Reply:
Here the `ifOpenService.open` was initialized and thats why I checked from `undefined` to `false` but I can see how this can be confusing. Changing to test from `true` to `false`